src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -932,46 +934,66 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		this._terminalAltBufferActiveContextKey.set(!!(this._xterm && this._xterm.buffer.active === this._xterm.buffer.alternate));
 	}
 
-	override dispose(immediate?: boolean): void {
-		this._logService.trace(`terminalInstance#dispose (instanceId: ${this.instanceId})`);
-		dispose(this._linkManager);
-		this._linkManager = undefined;
-		dispose(this._commandTrackerAddon);
-		this._commandTrackerAddon = undefined;
-		dispose(this._widgetManager);
+	override dispose(immediate?: boolean, confirmed?: boolean): void {
```

Comment:
the below is all indented bc I can't return a promise (IE `return this._disposeAsync`)

Reply:
I think we should be doing this at the service level for the following reasons:

- dispose should only get called once where as this will kind of bail out before `super.dispose()` is called. I've added parameters to dispose elsewhere and it felt like a hack
- I'm worried about hard to find regressions elsewhere because of the amount of ways a terminal can be taken down.

So something like `TerminalService.safeDisposeTerminal(): Promise<boolean>` which we would call anywhere we want to confirm